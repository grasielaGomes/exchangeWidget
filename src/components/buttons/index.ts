export * from "./FullButton";
export * from "./TextButton";