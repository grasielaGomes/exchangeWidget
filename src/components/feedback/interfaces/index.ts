
export interface FeedbackI {
  errorMessage?: string;
  isSuccess: boolean;
  successMessage?: string;
}