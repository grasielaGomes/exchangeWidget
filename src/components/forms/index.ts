import { DropdowMenu } from './DropdowMenu';
export { DropdowMenu}