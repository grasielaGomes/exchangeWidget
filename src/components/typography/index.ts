export * from "./Heading";
export * from "./CustomText";