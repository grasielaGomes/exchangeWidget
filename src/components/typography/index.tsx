import { HeadingI, TextI } from "./interfaces";

export const Heading = ({ color = "", children, variant = "h2" }: HeadingI) => {
  const styleBase = "font-bold";
  const variants = {
    h1: `${styleBase} text-2xl leading-9`,
    h2: `${styleBase} text-xl leading-8`,
    h3: `${styleBase} text-lg leading-7`,
  };
  return <p className={`${variants[variant]} ${color}`}>{children}</p>;
};

export const CustomText = ({
  color = "",
  children,
  variant = "body"
}: TextI) => {
  const styleBase = "font-normal";
  const variants = {
    body: `${styleBase} text-base leading-6`,
    small: `${styleBase} text-sm leading-5`,
    tiny: `${styleBase} text-xs leading-4`,
  };
  return <p className={`${variants[variant]} ${color}`}>{children}</p>;
};
