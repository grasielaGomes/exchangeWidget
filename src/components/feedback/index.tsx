import { CustomText } from "../typography";
import { FeedbackI } from "./interfaces/index";
import { useLayoutEffect, useState } from "react";

const texts = {
  errorMessage: "Server error. Please try again later.",
  successMessage: "Exchange submitted."
};

const styles = {
  successFeedback: (isSuccess: boolean, showFeedback: boolean) =>
    `fixed left-0 bottom-0 w-full flex items-center justify-center h-12 transition-opacity duration-200 ease-in-out opacity-0 ${
      isSuccess ? "bg-primary" : "bg-pink-500"
    } ${(showFeedback && "opacity-100") || "opacity-0"}`
};

export const Feedback = ({
  errorMessage = texts.errorMessage,
  isSuccess,
  successMessage = texts.successMessage
}: FeedbackI) => {
  const [showFeedback, setShowFeedback] = useState(isSuccess);

  useLayoutEffect(() => {
    setTimeout(() => {
      setShowFeedback(false);
    }, 3000);
  }, [isSuccess]);

  return (
    <div className={styles.successFeedback(isSuccess, showFeedback)}>
      <CustomText isBold color="white">
        {isSuccess ? successMessage : errorMessage}
      </CustomText>
    </div>
  );
};
