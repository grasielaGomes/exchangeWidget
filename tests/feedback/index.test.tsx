import { render } from "@testing-library/react";
import { Feedback } from "../../src/components/feedback";

describe("Tests on <Feedback />", () => {
  it("should match with snapshot on success", () => {
    const isSuccess = true;

    const { container } = render(<Feedback isSuccess={isSuccess} />);

    expect(container).toMatchSnapshot();
  });

  it("should show default error message", () => {
    const isSuccess = false;
    const errorMessage = "Server error. Please try again later.";

    const { getByText } = render(<Feedback isSuccess={isSuccess} />);

    expect(getByText(errorMessage)).toBeTruthy();
  });

  it("should show custom error message", () => {
    const isSuccess = false;
    const errorMessage = "Error message";

    const { getByText } = render(
      <Feedback isSuccess={isSuccess} errorMessage={errorMessage} />
    );

    expect(getByText(errorMessage).innerHTML).toBe(errorMessage);
  });
});
