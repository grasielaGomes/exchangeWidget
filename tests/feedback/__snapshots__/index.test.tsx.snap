// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Feedback /> should match with snapshot on success 1`] = `
<div>
  <div
    class="z-50 fixed left-0 bottom-0 w-full flex items-center justify-center h-12 transition-opacity duration-200 ease-in-out opacity-0 bg-primary opacity-100"
  >
    <p
      class="font-medium text-base leading-6 text-white"
    >
      Exchange submitted.
    </p>
  </div>
</div>
`;
